starship-1.2.0 fails to build on NetBSD

The new release fails to build on NetBSD with the following error.

```
Compiling starship v1.2.0 (/usr/pkgsrc/wip/starship/work/starship-1.2.0)
error[E0392]: parameter `'a` is never used
  --> src/context.rs:27:20
   |
27 | pub struct Context<'a> {
   |                    ^^ unused parameter
   |
   = help: consider removing `'a`, referring to it in a field, or using a marker such as `PhantomData`

For more information about this error, try `rustc --explain E0392`.
error: could not compile `starship` due to previous error
*** Error code 101

Stop.
bmake[1]: stopped in /usr/pkgsrc/wip/starship
*** Error code 1

Stop.
bmake: stopped in /usr/pkgsrc/wip/starship
error: External command failed
   ┌─ shell:22:1
   │
22 │ bmake
   │ ^^^^^ command failed
```

Features are disabled for this build due to non-existing support for the battery features.
See, https://github.com/svartalf/rust-battery/issues/20
Although, initial code to support these has been provided.

OS: NetBSD-9.99.93
Rust: 1.57.0

Regards,
pin

--- a/src/context.rs
+++ b/src/context.rs
@@ -15,6 +15,7 @@ use std::env;
 use std::ffi::{OsStr, OsString};
 use std::fmt::Debug;
 use std::fs;
+use std::marker::PhantomData;
 use std::num::ParseIntError;
 use std::path::{Path, PathBuf};
 use std::string::String;
@@ -70,6 +71,9 @@ pub struct Context<'a> {
 
     /// Starship root config
     pub root_config: StarshipRootConfig,
+
+    /// Avoid issues with unused lifetimes when features are disabled
+    _marker: PhantomData<&'a ()>,
 }
 
 impl<'a> Context<'a> {
@@ -149,6 +153,7 @@ impl<'a> Context<'a> {
             #[cfg(feature = "battery")]
             battery_info_provider: &crate::modules::BatteryInfoProviderImpl,
             root_config,
+            _marker: PhantomData,
         }
     }
 

